File: Desktop/glamai-fyp/mobile_app/lib/models/feedback.dart
class AppFeedback {
  final int id;
  final int appointmentId;
  final int userId;
  final int rating;
  final String reviewText;
  final String sentimentLabel;  // positive | neutral | negative
  final double sentimentScore;  // VADER compound -1.0 to 1.0
  final String? createdAt;

  const AppFeedback({
    required this.id,
    required this.appointmentId,
    required this.userId,
    required this.rating,
    required this.reviewText,
    required this.sentimentLabel,
    required this.sentimentScore,
    this.createdAt,
  });

  factory AppFeedback.fromJson(Map<String, dynamic> json) => AppFeedback(
        id: json['id'] as int,
        appointmentId: json['appointment_id'] as int,
        userId: json['user_id'] as int,
        rating: json['rating'] as int,
        reviewText: json['review_text'] as String,
        sentimentLabel: json['sentiment_label'] as String,
        sentimentScore: (json['sentiment_score'] as num).toDouble(),
        createdAt: json['created_at'] as String?,
      );
}

File: Desktop/glamai-fyp/backend/app/models/feedback.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship

from app.database import Base


class Feedback(Base):
    __tablename__ = "feedbacks"

    id = Column(Integer, primary_key=True, index=True)
    appointment_id = Column(Integer, ForeignKey("appointments.id"), nullable=False, unique=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    rating = Column(Integer, nullable=False)          # 1–5
    review_text = Column(String, nullable=False)
    sentiment_label = Column(String, nullable=False)  # positive | neutral | negative
    sentiment_score = Column(Float, nullable=False)   # VADER compound score (-1.0 to 1.0)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    appointment = relationship("Appointment", back_populates="feedback")
    user = relationship("User")
